### PR TITLE
Cleanup context.TODO(), migrate from DeprecatedCreateWorker to CreateWorker (3)

### PR DESCRIPTION
**Pull request by rfranzke**
**How to categorize this PR?**

/area dev-productivity
/kind cleanup

**What this PR does / why we need it**:
Similar to #3658 and #3681, this PR continues with migration a couple of our controllers from `controllerutils.DeprecatedCreateWorker` (deprecated since #1094) to `controllerutils.CreateWorker`. This helps to get rid of some remaining usages of `context.TODO()`. Also, it replace client-go informers/listers with informers from the controller-runtime.

**Which issue(s) this PR fixes**:
Part of #1648 

**Special notes for your reviewer**:
ℹ️ Once this PR is merged, I'll follow-up with more PRs that also cover the remaining/not-yet migrated controllers.
/squash

**Release note**:

```other operator
NONE
```

### CODE REVIEW

**Review by stoyanr — DISMISSED** [review]
/lgtm

File: pkg/controllermanager/controller/controllerregistration/controllerregistration_seed_control.go [hunk 1]
```diff
@@ -42,129 +39,92 @@ import (
...
+		if apierrors.IsNotFound(err) {
+			r.logger.Infof("Object %q is gone, stop reconciling: %v", request.Name, err)
+			return reconcile.Result{}, nil
+		}
+		r.logger.Infof("Unable to retrieve object %q from store: %v", request.Name, err)
+		return reconcile.Result{}, err
+	}
 
+	logger := logger.NewFieldLogger(r.logger, "controllerregistration-seed", seed.Name)
 	logger.Infof("[CONTROLLERINSTALLATION SEED] Reconciling %s", seed.Name)
```
Comment (ialidzhikov) [inline pkg/controllermanager/controller/controllerregistration/controllerregistration_seed_control.go]: Nit: you could omit the seed name when logging as the `logger` is now a field one.

```suggestion
	logger.Info("[CONTROLLERINSTALLATION SEED] Reconciling")
```

**Review by ialidzhikov — COMMENTED** [review]
I spotted only 1 nit.

**Review by ialidzhikov — DISMISSED** [review]
/lgtm

File: pkg/controllermanager/controller/controllerregistration/controllerregistration_control.go [hunk 2]
```diff
@@ -23,32 +23,31 @@ import (
...
+func (c *Controller) controllerRegistrationAdd(ctx context.Context, obj interface{}) {
 	key, err := cache.MetaNamespaceKeyFunc(obj)
 	if err != nil {
 		return
 	}
 	c.controllerRegistrationQueue.Add(key)
 
-	seedList, err := c.seedLister.List(labels.Everything())
-	if err != nil {
+	seedList := &gardencorev1beta1.SeedList{}
```
Comment (timebertt) [inline pkg/controllermanager/controller/controllerregistration/controllerregistration_control.go]: Should we use a metadata-only list here?

**Review by timebertt — COMMENTED** [review]
One suggestion.

General note: with this PR, we are doing a lot of direct get/list calls that were previously read from the cache. Though, I assume that this is not a problem, as gcm can communicate directly in-cluster with the garden API server.
When we perform similar refactorings in the gardenlet, we might have to be a bit careful.

File: pkg/controllermanager/controller/controllerregistration/controllerregistration_control.go (hunk 2 above)
Comment (rfranzke) [reply pkg/controllermanager/controller/controllerregistration/controllerregistration_control.go]: For sure, nice catch. Done with 14d2747

**Review by timebertt — APPROVED** [review]
Thanks for addressing my feedback and thanks for the PR in general!
/lgtm

**Pull request merged**